campaignion_overlay: HTML field for intro.

alex-mo opened the pull request.

alex-mo committed:
campaignion_overlay: HTML field for intro.

- add a generic class to trigger a close on click

alex-mo committed:
Fix only set form if there is a node loaded.

The target id could be "None"/NULL. In this case an "empty" form would
be loaded an hooks run, thus leading to undefined behaviour (e.g. form
alter hooks getting called with an incomplete form...)

torotil commented on campaignion_overlay/campaignion_overlay.module:
```diff
@@ -39,8 +39,12 @@ function campaignion_overlay_node_view($entity, $display, $view_mode) {
+        // only set form if a node is loaded
+        if ($node = node_load($content[0]['target_id'])) {
+          $form = drupal_get_form('webform_client_form_' . $node->nid, $node, []);
+        } else {
```
The `else` should be on its own line.

torotil commented on campaignion_overlay/campaignion_overlay.module:
```diff
@@ -57,7 +61,7 @@ function campaignion_overlay_node_view($entity, $display, $view_mode) {
           ],
           '#content' => $form,
-          '#introduction' => $intro[0]['safe_value'],
+          '#introduction' => field_view_field('field_collection_item', $overlay_options, 'campaignion_overlay_introduction',  array('label' => 'hidden')),
```
Please use the new-style array syntax.

torotil reviewed (changes requested):
Please run `phpcs` ;-)

torotil committed:
Migration: Coding-style fixes and streamlining.

torotil committed:
campaignion_overlay: Streamline rendering code.

torotil committed:
More streamlining.

- Use hook_field_collection_item_view_alter() so that there is no need
  to check for 'thank_you_page' anywhere.
- Use an entity_metadata_wrapper() for easier access to the
  field_collection’s fields.
- Don’t render enabled and content field: We only need their values.

torotil committed:
Add simple test for the overlay.
